esp/services/ws_smc/ws_smcService.cpp

```diff
@@ -2601,3 +2601,20 @@ bool CWsSMCEx::onLockQuery(IEspContext &context, IEspLockQueryRequest &req, IEsp
 
     return true;
 }
+
+bool CWsSMCEx::onGetBuildInfo(IEspContext &context, IEspGetBuildInfoRequest &req, IEspGetBuildInfoResponse &resp)
+{
+    try
+    {
+        context.ensureFeatureAccess(FEATURE_URL, SecAccess_Read, ECLWATCH_SMC_ACCESS_DENIED, SMC_ACCESS_DENIED);
+
+        if (isContainerized())
+            resp.setBuildInfo("CONTAINERIZED");
```

Comment:
Are you planning that this string may contain a list of flags in the future?  As opposed to defining explicit flags in your ESDL definition?

Reply:
You could make it an array of strings.  Or a collection of name value pairs.

Reply:
Either of those is better than defining it in your ESDL, but also wont break users when you add more.

Reply:
I thought that more flags may be added using comma separated BuildInfo. The name value pairs may be more flexible.

Reply:
yes the only problem with doing it the other way is that users may just do a string compare now, then later if you add another comma value it will break their string compare.  The user won't parse the string array or name value pair on their own.